src/app/dashboard/components/filters/form.tsx

```diff
@@ -1,47 +1,108 @@
-import { Button } from '@mui/material';
-import React from 'react';
+import { Dayjs } from 'dayjs';
 import { useForm, useWatch } from 'react-hook-form';
 import styled from 'styled-components';
+import { NumberParam, StringParam, useQueryParams } from 'use-query-params';
 
 import Form from 'components/form';
 import DateInput from 'components/form/dateInput';
+import NumberInput from 'components/form/numberInput';
 
 import { FiltersForm, FiltersFormSchema } from './schema';
-import { getFilterFormDefaultValues, minDateBoundary, today } from './utils';
+import { minDateBoundary, today } from './utils';
+import { useDebounce } from 'use-debounce';
+import { useEffect } from 'react';
+
+interface Props {
+  setStartDateHandle: (childData: Dayjs) => void;
+  setEndDateHandle: (childData: Dayjs) => void;
+  setAggregateHandle: (aggregate: number) => void;
+  startDate: Dayjs;
+  endDate: Dayjs;
+  aggregate: number;
+}
+
+const StockChartFiltersForm = (props: Props) => {
+  const [query, setQuery] = useQueryParams({
+    date_from: StringParam,
+    date_to: StringParam,
+    aggregate: NumberParam,
+  });
+  query;
 
-const StockChartFiltersForm: React.FC = () => {
   const methods = useForm<FiltersForm>({
-    defaultValues: getFilterFormDefaultValues(),
+    defaultValues: {
+      startDate: props.startDate,
+      endDate: props.endDate,
+      aggregate: props.aggregate,
+    },
     resolver: zodResolver(FiltersFormSchema),
   });
-  const { handleSubmit, control } = methods;
+  const { handleSubmit, control, setValue } = methods;
 
   const startDate = useWatch({ control, name: 'startDate' });
   const endDate = useWatch({ control, name: 'endDate' });
+  const aggregate = useWatch({ control, name: 'aggregate' });
+  const [debouncedAggregate] = useDebounce(aggregate, 500);
 
   const onSubmit = handleSubmit((data: FiltersForm) => {
-    console.log(data.startDate.format('YYYY-MM-DD'));
-    console.log(data.endDate.format('YYYY-MM-DD'));
+    data;
   });
 
+  function passStartDateToParent(date: Dayjs | null) {
+    if (date === null) return;
+    setValue('startDate', date);
+
+    props.setStartDateHandle(date);
+    setQuery({ date_from: convertISODatetimeWithTimezoneToISODate(date) });
+  }
+  passStartDateToParent;
+
+  function passEndDateToParent(date: Dayjs | null) {
+    if (date === null) return;
+    setValue('endDate', date);
+
+    props.setEndDateHandle(date);
+    setQuery({ date_to: convertISODatetimeWithTimezoneToISODate(date) });
+  }
+  passEndDateToParent;
+
+  function handleAggregateChange(
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
+  ) {
+    const newValue = event.target.value;
+    const aggregate = newValue === '' ? 1 : parseInt(newValue);
+    setValue('aggregate', aggregate);
+  }
+
+  useEffect(() => {
+    props.setAggregateHandle(debouncedAggregate);
+    setQuery({ aggregate: debouncedAggregate });
+  }, [debouncedAggregate]);
```

Comment:
5. Chcemy aktualizować `query` po jakiejkolwiek zmianie formularza, ale też tylko wtedy kiedy walidacja formularza przebiegnie pomyślnie
6. Zapewnia nam to też wrzucenie domyślnych wartości do URL podczas pierwszego wejścia na tą stronę np. po zalogowaniu lub przejściu z http://localhost:3000/app/statistics

```suggestion
   useEffect(() => {
    trigger().then((isValid) => {
      if (!isValid) return;

      setQuery({
        aggregate: debouncedAggregate,
        date_to: endDate.format('YYYY-MM-DD'),
        date_from: startDate.format('YYYY-MM-DD'),
      });
    });
  }, [debouncedAggregate, startDate, endDate]);
```

Reply:
fixed